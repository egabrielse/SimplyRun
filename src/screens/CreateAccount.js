import React, { Component } from 'react';
import { Text, View, TextInput, Button, Alert } from 'react-native';
import firebase from 'firebase'
import firebaseConfig from '../config/firebaseConfig'
import '@firebase/firestore';
import { connect } from 'react-redux'


//Firebase initialization 
firebaseConfig


class CreateAccount extends Component {

    signUp = () => {
        if (this.props.email != null && this.props.password != null &&
            this.props.email.trim() != "" && this.props.password.trim() != "") {
            firebase
                .auth()
                .createUserWithEmailAndPassword(this.props.email, this.props.password)
                .then(() => this.props.navigation.navigate('Main'))
                .catch(error => Alert.alert(error.message));
        }
    }
    render() {
        return (
            <View style={{ flex: 1, alignContent: 'center' , justifyContent: 'center' }} >
                <Text>Sign Up</Text>
                <TextInput
                    placeholder="Email"
                    autoCapitalize="none"
                    onChangeText={email=> this.props.typeEmail(email)}
                    value={this.props.email}
                />
                 <TextInput
                    secureTextEntry
                    placeholder="Password"
                    autoCapitalize="none"
                    onChangeText={password => this.props.typePassword(password)}
                    value={this.props.password}
                />
                <Button title="Sign Up" onPress={this.signUp} />
     
            </View>
        );
    }
}


//Getting the states from the store and mapping it to props in the CreateAccountScreen
function mapStateToProps(state) {
    return {
        email: state.email,
        password: state.password
    }
}
//Sends actions to the reducer in the App.js file 
function mapDispatchtoProps(dispatch){
    return {
        typeEmail: (email) => dispatch({ type: "EMAIL", email }),
        typePassword: (password) => dispatch({ type: "PASSWORD", password })
    }
}

export default connect(mapStateToProps, mapDispatchtoProps)(CreateAccount);