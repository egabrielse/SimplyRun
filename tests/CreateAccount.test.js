import React from 'react';
import CreateAccount from '../src/screens/CreateAccount';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import initialState from './mock_stores/initialState'
import { Provider } from 'react-redux';

const createMockStore = configureStore([])

describe("CreateAccount screen", () => {

    beforeEach(() => {
        mockStore = createMockStore(initialState)
    })

    it('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={mockStore}>
                <CreateAccount/>
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})
