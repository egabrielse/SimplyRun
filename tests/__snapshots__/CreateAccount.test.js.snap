// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount screen renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "marginHorizontal": 20,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 40,
              "fontStyle": "italic",
              "fontWeight": "bold",
            }
          }
        >
          Simply Run
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "borderColor": "lightgrey",
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "maxHeight": 50,
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
          value={null}
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          returnKeyType="next"
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "lightgrey",
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "maxHeight": 50,
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
          value={null}
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Confirm Password"
          rejectResponderTermination={true}
          returnKeyType="done"
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "lightgrey",
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "maxHeight": 50,
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
          value={null}
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "lightgray",
                "height": 50,
                "justifyContent": "center",
                "paddingHorizontal": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                }
              }
            >
              Sign Up!
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "red",
                "height": 50,
                "justifyContent": "center",
                "maxHeight": 50,
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              DEV:Skip Account
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
